### PR TITLE
Add slot count for Snack

### PR DESCRIPTION
**Pull request by RamZallan**
* Add count column to Slot
* Don't drop from Slots w/ count < 1
* Set Slot inactive once its last item was dropped

### CODE REVIEW

File: mizu/drinks.py [hunk 1]
```diff
@@ -142,7 +143,7 @@ def drop_drink(adapter, user = None):
     logger.debug('Drop request is valid')
 
     slot_status = _get_machine_status(body['machine'])
-    if slot_status[body['slot']-1]['empty']:  # slots are 1 indexed
+    if body['machine'] == 'snack' and slot.count < 1 or slot_status[body['slot']-1]['empty']:  # slots are 1 indexed
```
Comment (hearthsinger) [inline mizu/drinks.py]: Would check to make sure that this doesn't short circuit out of the `and` if `body['machine'] != 'snack'` - maybe toss some parens in here just to be careful idk.

File: mizu/drinks.py [hunk 2]
```diff
@@ -198,6 +199,12 @@ def drop_drink(adapter, user = None):
     _manage_credits(user['preferred_username'], new_balance, adapter)
     logger.debug('Credits for {} updated'.format(user['preferred_username']))
 
+    if body['machine'] == 'snack':
+        slot.count = Slot.count - 1 # Decrement stock count, set inactive if empty
```
Comment (hearthsinger) [inline mizu/drinks.py:203]: Okay now I am confused here, shouldn't both of these be lowercase `slot`?

File: mock.json [hunk 3]
```diff
@@ -1 +1 @@
...
...r": 3, "item": 7, "active": true, "count": 0}, {"machine": 1, "number": 6, "item": 15, "active": false, "count": 0}, {"machine": 1, "number": 7, "item": 15, "active": false, "count": 0}, {"machine": 2, "number": 1, "item": 8, "active": false, "count": 0}, {"machine": 1, "number": 4, "item": 14, "active": true, "count": 0}, {"machine": 1, "number": 5, "item": 22, "active": true, "count": 0}, {"machine": 1, "number": 1, "item": 6, "active": true, "count": 0}], "Items": [{"id": 4, "name": "Dr. Pepper", "price": 10}, {"id": 5, "name": "La Croix", "price": 10}, {"id": 6, "name": "Grape Fanta", "price": 10}, {"id": 7, "name": "Jolt", "price": 200}, {"id": 12, "name": "Sprite", "price": 10}, {"id": 14, "name": "7-up", "price": 10}, {"id": 10, "name": "Coke Zero", "price": 10}, {"id": 8, "name": "Bawls", "price": 100}, {"id": 18, "name": "New Beverage", "price": 69}, {"id": 19, "name": "Test", "price": 100}, {"id": 20, "name": "Foo", "price": 9}, {"id": 1, "name": "Coke", "price": 10}, {"id": 21, "name": "Conke", "price": 42069}, {"id": 22, "name": "Drink's Choice", "price": 5}, {"id": 15, "name": "Empty", "price": 99999}], "Users": [{"cn": "Keith", "uid": "keith", "drinkBalance": 42069}]}
```
Comment (hearthsinger) [inline mock.json:1]: lol

**Review by zthart — APPROVED** [review]
lgtm pending comments

File: mizu/drinks.py (hunk 2 above)
Comment (RamZallan) [reply mizu/drinks.py:203]: Not really sure, I was going off of this:

https://stackoverflow.com/questions/2334824/how-to-increase-a-counter-in-sqlalchemy

File: mock.json (hunk 3 above)
Comment (RamZallan) [reply mock.json:1]: yeah i just added the counts = 0

File: mizu/drinks.py (hunk 2 above)
Comment (hearthsinger) [reply mizu/drinks.py:203]: Oh I see, it's some fuck shit with the operation being atomic in the db but not atomic if you eval'd `slot.count` in Python then subtracted. Either it works or we get a 500 so whatever works.

**Pull request merged**